### ISSUE TITLE
Add more profile metadata fields

### ISSUE THREAD

**Issue by davispuh** [mastodon/mastodon#29906]
### Pitch

Currently in your Profile you can add 4 metadata fields but that is way too few.
You might want to link to your other social profiles and such.

### Motivation

Basically I am looking for open source replacement of https://about.me/ and Mastodon almost covers it but I need more fields to add more links.

This is how it looks like on https://about.me/

![about](https://github.com/mastodon/mastodon/assets/651800/58892b31-eaa1-454e-984e-ae157dcd3e75)

and on Discord

![discord](https://github.com/mastodon/mastodon/assets/651800/44f95176-023c-4143-95a3-39b93c2539d1)

So basically want something similar in Mastodon's profile page.

**Comment by timothyjrogers** [comment]
Looking at the code around this it seems like it would be fairly straight forward to allow an arbitrary number of fields in a user's profile. They're just stored in a JSONB field in the account table, there's no reason that couldn't support a huge number of fields. The number of fields (4) is currently hardcode in the [account controller](https://github.com/mastodon/mastodon/blob/main/app/models/account.rb#L67).

The concern that immediately occurs to me is that allowing arbitrarily large account profiles might have impacts to instances with a lot of users or instances that are running with lean hardware/resources. Rather than allowing arbitrarily many account fields maybe the number could be configured per-instance as a configuration item.

Assuming the above is a meaningful concern, that would then raise the usability question of how to handle this when migrating instance. Specifically, what happens if I migrate from an instance with a limit of, say, 10 fields to an instance that only allows 4.

**Comment by davispuh** [comment]
I don't think it's good idea to have it configurable, just hardcode to like 20 which would be plenty for everyone and won't cause any performance issues since it's really nothing. And regarding usability that's on each user itself if their profile is unreadable etc.

Also maybe better would be to have dedicated fields for links. Having icons for them like you can see on Discord and show additional metadata aswell like Karma and Followers seems very nice. What icon to use and additional info could be inferred from domain name. Well maybe some people would want custom icons and stuff.

**Comment by floe** [comment]
+1 for this feature. One option to get started on this would be to keep the publicly visible fields limited to four (keeping the UI as-is) but enable a larger number of fields in the backend, perhaps something like 32? Also, I'd suggest to raise the cap on field length which IIRC is currently 256 - this would enable a bunch of other use cases as well, e.g. use it as an ultra-lightweight data storage for other apps. I'm thinking of something like "store my highscores from the crossword and also let my followers see them if they're interested".
